src/main/java/in/org/projecteka/hiu/dataflow/DataFlowService.java

```diff
@@ -20,4 +26,27 @@
     private Mono<Void> insertHealthInformation(Entry entry, String transactionId) {
         return dataFlowRepository.insertHealthInformation(transactionId, entry);
     }
-}
+
+    public Flux<DataEntry> fetchHealthInformation(String consentRequestId, String requesterId) {
+        return consentRepository.getConsentDetails(consentRequestId)
+                .flatMap(consentDetail -> {
+                    if (consentDetail.get("requester").equals(requesterId))
```

Comment:
You can use filter for filtering consentDetail.get("requester").equals(requesterId) and do flatMap with switchIfEmpty and throw an error.